tidb-binlog/pump_client/client.go

```diff
@@ -321,42 +342,42 @@ func (c *PumpsClient) WriteBinlog(binlog *pb.Binlog) error {
 	log.Info("[pumps client] write binlog to avaliable pumps all failed, will try unavaliable pumps")
 	pump, err1 := c.backoffWriteBinlog(req, binlog.Tp, binlog.StartTs)
 	if err1 == nil {
+		choosePump = pump
 		return nil
 	}
-	choosePump = pump
 
 	return errors.Errorf("write binlog failed, the last error %v", err)
 }
 
+// return directly for non p-binlog
+// try every online pump for p-binlog
 func (c *PumpsClient) backoffWriteBinlog(req *pb.WriteBinlogReq, binlogType pb.BinlogType, startTS int64) (pump *PumpStatus, err error) {
 	if binlogType != pb.BinlogType_Prewrite {
 		// never return error for commit/rollback binlog.
 		return nil, nil
 	}
 
-	unAvaliablePumps := make([]*PumpStatus, 0, 3)
+	allPumps := make([]*PumpStatus, 0, 3)
 	c.RLock()
-	for _, pump := range c.Pumps.UnAvaliablePumps {
-		unAvaliablePumps = append(unAvaliablePumps, pump)
+	for _, pump := range c.Pumps.Pumps {
+		allPumps = append(allPumps, pump)
```

Comment:
Can reuse `copyPumps` here.

Reply:
argument is different

Reply:
Is `allPumps := copyPumps(c.Pumps.Pumps)` OK?